support-frontend/assets/helpers/customHooks/useAnalyticsProfile.ts

```diff
@@ -1,80 +1,81 @@
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { fetchJson } from 'helpers/async/fetch';
 import * as cookie from 'helpers/storage/cookie';
-import { getUser } from 'helpers/user/user';
 
-const useAnalyticsProfile = () => {
-	const user = getUser();
+const COOKIE_NAME = 'GU_user_analytics_profile';
+const COOKIE_TTL_DAYS = 1;
 
+const useAnalyticsProfile = () => {
 	const [hasMobileAppDownloaded, setHasMobileAppDownloaded] = useState(false);
 	const [hasFeastMobileAppDownloaded, setHasFeastMobileAppDownloaded] =
 		useState(false);
+	const [dataLoaded, setDataLoaded] = useState(false);
 
-	useEffect(() => {
-		const COOKIE_NAME = 'GU_user_analytics_profile';
-		const COOKIE_TTL_DAYS = 1;
-
-		const loadAnalyticsData = async () => {
-			if (!user.isSignedIn) {
-				return;
-			}
+	const loadAnalyticsData = useCallback(async () => {
+		if (dataLoaded) {
+			return;
+		}
 
-			const cachedData = cookie.get(COOKIE_NAME);
+		const cachedData = cookie.get(COOKIE_NAME);
 
-			if (cachedData) {
-				try {
-					const parsedData = JSON.parse(cachedData) as {
-						hasMobileAppDownloaded?: boolean;
-						hasFeastMobileAppDownloaded?: boolean;
-					};
+		if (cachedData) {
+			try {
+				const parsedData = JSON.parse(cachedData) as {
+					hasMobileAppDownloaded?: boolean;
+					hasFeastMobileAppDownloaded?: boolean;
+				};
 
-					setHasMobileAppDownloaded(parsedData.hasMobileAppDownloaded ?? false);
-					setHasFeastMobileAppDownloaded(
-						parsedData.hasFeastMobileAppDownloaded ?? false,
-					);
-					return;
-				} catch (error) {
-					console.error('Error parsing cached analytics data:', error);
-					cookie.set(COOKIE_NAME, '', -1);
-				}
+				setHasMobileAppDownloaded(parsedData.hasMobileAppDownloaded ?? false);
+				setHasFeastMobileAppDownloaded(
+					parsedData.hasFeastMobileAppDownloaded ?? false,
+				);
+				setDataLoaded(true);
+				return;
+			} catch (error) {
+				console.error('Error parsing cached analytics data:', error);
+				cookie.set(COOKIE_NAME, '', -1);
 			}
+		}
 
-			try {
-				const response = await fetchJson<{
-					identityId: string;
-					hasMobileAppDownloaded: boolean;
-					hasFeastMobileAppDownloaded: boolean;
-				}>('/analytics-user-profile', {
-					mode: 'cors',
-					credentials: 'include',
-				});
+		try {
+			const response = await fetchJson<{
```

Comment:
it looks like if there was any client side or backend issue, it would try to call mparticle on every render, which could blow our quota pretty quickly.  I think it should make one attempt, and then not try again on the given onboarding journey.